MPSGameLift/Code/Source/Components/UI/UiGameLiftConnectWithPlayerSessionData.cpp

```diff
@@ -110,35 +109,51 @@ namespace MPSGameLift
             return;
         }
 
-        if (!document.HasMember("GameSessionId"))
+        // Alert the user if any information is missing from the JSON they provided
+        if (!document.HasMember("PlayerSessionId"))
         {
-            UiTextBus::Event(m_jsonParseFailTextUi, &UiTextInterface::SetText, "Missing GameSessionId!");
+            UiTextBus::Event(m_jsonParseFailTextUi, &UiTextInterface::SetText, "Missing PlayerSessionId!");
             return;
         }
 
-        if (!document.HasMember("PlayerSessionId"))
+        if (!document.HasMember("IpAddress") && !document.HasMember("DnsName"))
         {
-            UiTextBus::Event(m_jsonParseFailTextUi, &UiTextInterface::SetText, "Missing PlayerSessionId!");
+            UiTextBus::Event(m_jsonParseFailTextUi, &UiTextInterface::SetText, "Must provide either an IpAddress or DnsName!");
             return;
         }
```

Comment:
is it not easier to connect via game session id?  won't you not know the server IP or DNS name until the fleet is initialized or something and you look for that info in the console?

Reply:
Easier, yes. There’s a AWS GameLift gem API to connect by passing a game session id. That method will ask game lift to give back a player session, dns, and port and then does what my function is doing behind the scenes. For the 5-Click-Sample the developer will have generated the player session id themselves using the GameLift website.